Adding new section: Documentation style guide (#103)
mattermost/mattermost-handbook: operations/operations/publishing/publishing-guidelines/documentation-style-guide.md

presented with a style and tone that is appropriate for technical content, and easy to navigate.

## Document structure

Structure and organization are an important part of a document's ease of use and its understandability. 
Information should be organized and presented in a logical order, with similar subjects grouped together in the 
same section. In most cases, a document has a title, an introductory paragraph, and one or more sections.

Try to keep only one topic in a page. Shorter topics are easier to reuse in other documents, are 
easier to write and edit, and are easier to translate.

### Document sections

To make pages easier for people to quickly scan for the content that they're looking for, break your 
document up into logical sections. Each section should have a title, and the title should relate 
to the content of the section. Use sentence case for the section title.

A section title is not required if you have only one section.

### Document title

Use a title that accurately reflects the content of the document. People scan the table of contents 
looking for answers; it's often faster than using the built-in search engine.

> Use title case for document titles. For more information and an example of capitalization, see :ref:`capital`.

Reviewer: Not sure how to check if the reference link works.

Author: Good point. I'd normally do a local build to check but realized that it doesn't work the same as the Sphinx docs. I'll remove for now.

> Use title case for document titles. For more information and an example of capitalization, see :ref:`capital`.

Author: ```suggestion
Use title case for document titles (e.g. "This is an article about documentation").
```

## Using ReStructuredText markup (rst)

The reStructuredText specification allows for a certain degree of flexibility in 
markup to achieve your goals. For example, you can use any one of more than a dozen characters for 
section title underlines, and you have the option of using an overline in addition to an underline.

The majority of Mattermost technical documentation is written in `.rst`. However, there are some instances
where Markdown is used. 

Use the following conventions when writing in `.rst`. 

### Page titles

Underline page titles using `=`, with no overline. Underlines should be as long as the title text. For example:

```
  Document title
  ==============
```

### Section titles

Underline using `-` for section titles. For example:

```
  Section title
  -------------
``` 

Underline subsections using `~` for the first subsection level, and `^` for the second subsection level. For example:

```
  Subsection one
  ~~~~~~~~~~~~~~

  Subsection two
  ^^^^^^^^^^^^^^
``` 

### Table of contents

Insert a table of contents into a document using the following format:

```
  .. contents::
      :backlinks: top
```  
    
### Inline Markup

> Bold

Reviewer: The "Bold", etc. headings in this section currently render together with the description.

### Bullet lists

For bullet lists and sublists, use `-` before the list item. For example:

```
  - list item one
  - list item two
    - sublist item one
    - sublist item two
  - list item three
```

### Numbered lists and procedure steps

Create numbered lists and procedure steps using numbers for the top-level list and lower case alpha characters for the first nested list. For example:

```
  1. This is item one.
  2. This is item two.

    a. This is nested item one.
    b. This is nested item two.

  3. This is item three.
```

### Name-value groups

To create a name-value group such as a definition list, type the term on a line by itself. On the next line, indent the definition. For example:

```
  Total users
    The total number of active accounts created on your system. Excludes inactive accounts.
  Total teams
    The total number of teams created on your system.
```

### External links

URLs are automatically rendered as links in Sphinx; however, where possible, it is preferred that hyperlinks are created within the text of a sentence. Hyperlinks within a sentence can be created using the following formatting:

> `Link display text <URL-of-website>`__`, for example:

Reviewer: ```suggestion
`Link display text <URL-of-website>`__, for example:
```

Not sure if the last backstick is needed or if the intention is to have another backstick at the beginning.

The majority of Mattermost technical documentation is written in `.rst`. However, there are some instances
where Markdown is used. 

Use the following conventions when writing in `.rst`. 

### Page titles

Underline page titles using `=`, with no overline. Underlines should be as long as the title text. For example:

```
  Document title
  ==============
```

### Section titles

Underline using `-` for section titles. For example:

```
  Section title
  -------------
``` 

Underline subsections using `~` for the first subsection level, and `^` for the second subsection level. For example:

```
  Subsection one
  ~~~~~~~~~~~~~~

  Subsection two
  ^^^^^^^^^^^^^^
``` 

### Table of contents

Insert a table of contents into a document using the following format:

```
  .. contents::
      :backlinks: top
```  
    
### Text formatting
-----------------

Use highlighting of text to visually set off words and phrases that are important to readers. Content that should be highlighted includes file names, UI controls, and window titles. The following table has a comprehensive list with examples.

| Text          | Format        | Example             |

> | ------------- |:-------------:| -------------------:|

Reviewer: ```suggestion
| ------------- |:-------------:| --------------------|
```

Not sure if the "Example" heading is on the RHS because of this or if it's going to look ok when merged without any changes.
